packages/legend-application-studio/src/stores/editor/NewElementState.ts

```diff
@@ -471,23 +479,52 @@ export class NewPackageableConnectionDriver extends NewElementDriver<Packageable
 
 export class NewServiceDriver extends NewElementDriver<Service> {
   mappingOption?: PackageableElementOption<Mapping> | undefined;
+  runtimeOption?: RuntimeOption | undefined;
   constructor(editorStore: EditorStore) {
     super(editorStore);
 
     makeObservable(this, {
       mappingOption: observable,
+      runtimeOption: observable,
       setMappingOption: action,
+      setRuntimeOption: action,
+      getCompatibleRuntimeOptions: action,
       isValid: computed,
       createElement: action,
     });
     this.mappingOption =
       editorStore.graphManagerState.usableMappings.map(buildElementOption)[0];
+    this.runtimeOption = this.getCompatibleRuntimeOptions(
+      this.mappingOption?.value,
+    )[0];
   }
 
   setMappingOption(val: PackageableElementOption<Mapping> | undefined): void {
     this.mappingOption = val;
   }
 
+  setRuntimeOption(val: RuntimeOption | undefined): void {
+    this.runtimeOption = val;
+  }
+
+  getCompatibleRuntimeOptions(mapping: Mapping | undefined): RuntimeOption[] {
```

Comment:
this should be a computed attribute and change when mapping changes.

Reply:
<img width="397" alt="Screenshot 2023-09-21 at 5 00 13 PM" src="https://github.com/finos/legend-studio/assets/73408381/7aa8bc87-6e74-46a9-86ed-ae5f3e59a419">
already marked it as action 
the dropdown changes every time the selected mapping is changed

Reply:
updated code to be computed